rpc/src/test/scala/org/bitcoins/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,517 @@
+                    assert(
+                      transaction
+                        .inputs(0)
+                        .previousOutput
+                        .txId
+                        .flip == input0.txId)
+                    assert(
+                      transaction
+                        .inputs(1)
+                        .previousOutput
+                        .txId
+                        .flip == input1.txId)
+                  }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private def createRawTransaction: Future[Transaction] = {
+    client.generate(2).flatMap { blocks =>
+      client.getBlock(blocks(0)).flatMap { block0 =>
+        client.getBlock(blocks(1)).flatMap { block1 =>
+          client.getTransaction(block0.tx(0)).flatMap { transaction0 =>
+            client.getTransaction(block1.tx(0)).flatMap { transaction1 =>
+              val input0 =
+                TransactionOutPoint(transaction0.txid,
+                                    UInt32(transaction0.blockindex.get))
+              val input1 =
+                TransactionOutPoint(transaction1.txid,
+                                    UInt32(transaction1.blockindex.get))
+              val sig: ScriptSignature = ScriptSignature.empty
+              client.getNewAddress().flatMap { address =>
+                client.createRawTransaction(
+                  Vector(TransactionInput(input0, sig, UInt32(1)),
+                         TransactionInput(input1, sig, UInt32(2))),
+                  Map((address.value, Bitcoins(0.0000001))))
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  it should "be able to decode a raw transaction" in {
+    val transactionF = createRawTransaction
+    transactionF.flatMap { transaction =>
+      client.decodeRawTransaction(transaction).map { rpcTransaction =>
+        assert(rpcTransaction.txid.flip == transaction.txId)
+        assert(rpcTransaction.locktime == transaction.lockTime)
+        assert(rpcTransaction.size == transaction.size)
+        assert(rpcTransaction.version == transaction.version.toInt)
+        assert(rpcTransaction.vsize == transaction.vsize)
+      }
+    }
+  }
+
+  it should "be able to sign a raw transaction with wallet keys" in {
+    val transactionF = createRawTransaction
+    transactionF.flatMap { transaction =>
+      client.signRawTransactionWithWallet(transaction).map {
+        signedTransaction =>
+          logger.info(signedTransaction.toString)
+          logger.info(signedTransaction.hex.hex)
+          assert(signedTransaction.complete)
+      }
+    }
+  }
+
+  it should "be able to send a raw transaction to the mem pool" in {
+    val transactionF = createRawTransaction
+    transactionF.flatMap { transaction =>
+      client.signRawTransactionWithWallet(transaction).flatMap {
+        signedTransaction =>
+          client
+            .generate(100)
```

Comment:
Instead of generating 100 blocks in every test can we just generate 150 blocks in the `beforeAll` section of the test suite?